scripts/create_cloud_image.py

```diff
@@ -70,25 +90,47 @@ def main():
         if image.name == args.imagename
     ]
     current_ids = [image.id for image in current_images]
+    print("Existing image IDs: {}".format(current_ids))
 
     conn.compute.create_server_image(
         server=args.serveruuid, name=args.imagename
     )
-
-    for image in conn.image.images():
-        if image.name == args.imagename and image.id not in current_ids:
-            saved_image = image
+    for _ in xrange(240):
```

Comment:
4 hours to find the image id after creation?

Reply:
Yeah, due to the size of the image and the fact that it's asynchronous, I remeber @hughsaunders mentioning that sometimes it takes a very long time.